samples/Ark.ReferenceProject/Core/Ark.Reference.Core.Tests/Features/BookPrintProcessSteps.cs

```diff
@@ -0,0 +1,160 @@
+        }
+
+        [Given(@"I have created a book print process for that book with ShouldFail (.*)")]
+        public void GivenIHaveCreatedABookPrintProcessWithShouldFail(bool shouldFail)
+        {
+            var request = new BookPrintProcess.V1.Create
+            {
+                BookId = _currentBookId!.Value,
+                ShouldFail = shouldFail
+            };
+
+            _client.PostAsJson(_controllerName, request);
+            _client.ThenTheRequestSucceded();
+                
+            var response = _client.ReadAs<BookPrintProcess.V1.Output>();
+            _currentPrintProcessId = response.BookPrintProcessId;
+        }
+
+        [When(@"I try to create another book print process for that book")]
+        public void WhenITryToCreateAnotherBookPrintProcessForThatBook()
+        {
+            var request = new BookPrintProcess.V1.Create
+            {
+                BookId = _currentBookId!.Value,
+                ShouldFail = false
+            };
+
+            _client.PostAsJson(_controllerName, request);
+        }
+
+        [When(@"I retrieve the print process status")]
+        public void WhenIRetrieveThePrintProcessStatus()
+        {
+            _client.Get($"{_controllerName}/{_currentPrintProcessId}");
+        }
+
+        [When(@"I wait background bus to idle and outbox to be empty")]
+        public async Task WhenIWaitBackgroundBusToIdleAndOutboxToBeEmpty()
+        {
+            await _testHost.ThenIWaitBackgroundBusToIdleAndOutboxToBeEmpty().ConfigureAwait(false);
+        }
+
+        [Then(@"the print process should be created with status ""(.*)""")]
+        public void ThenThePrintProcessShouldBeCreatedWithStatus(string status)
+        {
+            _client.ThenTheRequestSucceded();
+            _currentPrintProcess = _client.ReadAs<BookPrintProcess.V1.Output>();
+            _currentPrintProcessId = _currentPrintProcess.BookPrintProcessId;
+            
+            _currentPrintProcess.Status.ToString().Should().Be(status);
+        }
+
+        [Then(@"the print process progress should be (.*)")]
+        public void ThenThePrintProcessProgressShouldBe(double expectedProgress)
+        {
+            if (_currentPrintProcess == null && _client.LastStatusCodeIsSuccess())
+            {
+                _currentPrintProcess = _client.ReadAs<BookPrintProcess.V1.Output>();
+            }
+            
+            _currentPrintProcess.Should().NotBeNull();
+            _currentPrintProcess!.Progress.Should().BeApproximately(expectedProgress, 0.01);
+        }
+
+        [Then(@"the business rule violation code is ""(.*)""")]
+        public void ThenTheBusinessRuleViolationCodeIs(string expectedCode)
+        {
+            var problemDetails = _client.ReadAs<ProblemDetails>();
+            problemDetails.Should().NotBeNull();
+            problemDetails.Title.Should().NotBeNullOrEmpty();
+            problemDetails.Extensions.Should().ContainKey("errorCode");
```

Comment:
this is strange. the specific BusinessRuleViolation that has been created should be expected and Title comparison should be jsed.